big_scape/output/legacy_output.py

```diff
@@ -1146,8 +1154,452 @@ def legacy_generate_bin_output(
+            gbk_path = DB.execute(select_statment).fetchone()[0]
+
+            row = "\t".join(
+                [
+                    str(gbk_path),
+                    record_type,
+                    str(record_number),
+                    str(gcf_number),
+                ]
+            )
+            clustering_file.write(row + "\n")
+
+    return None
+
+
+def write_network_file(
+    network_file_path: Path,
+    existing_distances: set[
+        tuple[int, int, float, float, float, float, str, int, int, int, int, str]
+    ],
+) -> None:
+    """Writes a network file to the output directory
+
+    Args:
+        network_file_path (Path): path to write file to
+        existing_distances (set): edgelist to populate file
+
+    Raises:
+        RuntimeError: no db present
+
+    """
+
+    if not DB.metadata:
+        raise RuntimeError("DB.metadata is None")
+
+    gbk_table = DB.metadata.tables["gbk"]
+    bgc_record_table = DB.metadata.tables["bgc_record"]
+
+    with open(network_file_path, "w") as network_file:
+        header = "\t".join(
+            [
+                "GBK_a",
+                "Record_Type_a",
+                "Record_Number_a",
+                "GBK_b",
+                "Record_Type_b",
+                "Record_Number_b",
+                "distance",
+                "jaccard",
+                "adjacency",
+                "dss",
+                "weights",
+                "aligmnent_mode",
+                "ORF_coords_a",
+                "ORF_coords_b",
+            ]
+        )
+
+        network_file.write(header + "\n")
+
+        for dist in existing_distances:
+            (
+                region_a_id,
+                region_b_id,
+                distance,
+                jaccard,
+                adjacency,
+                dss,
+                weights,
+                ext_a_start,
+                ext_a_stop,
+                ext_b_start,
+                ext_b_stop,
+                alignment_mode,
+            ) = dist
+
+            # for record A
+            select_statment = select(
+                bgc_record_table.c.gbk_id,
+                bgc_record_table.c.record_number,
+                bgc_record_table.c.record_type,
+            ).where(bgc_record_table.c.id == region_a_id)
+            gbk_id_a, record_number_a, record_type_a = DB.execute(
+                select_statment
+            ).fetchone()
+
+            select_statment = select(gbk_table.c.path).where(gbk_table.c.id == gbk_id_a)
+            gbk_path_a = DB.execute(select_statment).fetchone()[0]
+
+            # for record B
+            select_statment = select(
+                bgc_record_table.c.gbk_id,
+                bgc_record_table.c.record_number,
+                bgc_record_table.c.record_type,
+            ).where(bgc_record_table.c.id == region_b_id)
+            gbk_id_b, record_number_b, record_type_b = DB.execute(
+                select_statment
+            ).fetchone()
```

Comment:
challenge if you want it: there is a way to join these two select statements as one.

I think if we want to optimize it further, there should also be a way to cut out the middleman of passing existing_distances.

both of these can be TODO items as chores. they may be needed for very large datasets.

Reply:
added as TODOs for now